opus/application/static_media/js/sortMetadata.js

```diff
@@ -0,0 +1,280 @@
+        let top =  e.pageY;
+        let left = ($(tab).innerWidth() - e.pageX > menu.width)  ? e.pageX + 12: e.pageX-menu.width;
+
+        $(contextMenu).css({
+            display: "block",
+            top: top,
+            left: left
+        }).addClass("show");
+    },
+
+    // update order arrows right away when user clicks on sorting arrows in pill or table header
+    // sync up arrows in both sorting pill and table header
+    updateOrderIndicator: function(headerOrderIndicator, pillOrderIndicator, isDescending, slug) {
+        let headerOrder = isDescending ? "desc" : "asc";
+        let headerOrderArrow = isDescending ? tableSortUpArrow : tableSortDownArrow;
+
+        // If header already exists, we update the header arrow, else we do nothing
+        if (headerOrderIndicator && headerOrderIndicator.length !== 0) {
+            headerOrderIndicator.data("sort", `${headerOrder}`);
+            headerOrderIndicator.attr("class", `op-column-ordering ${headerOrderArrow}`);
+        }
+        $(pillOrderIndicator).parent().data("descending", (isDescending ? "true" : "false"));
+    },
+
+    updateSortOrder: function(data) {
+        let tab = opus.getViewTab();
+        let listHtml = "";
+
+        let tableColumnFields = {};
+        $(`${tab} .op-data-table-view th`).find("a[data-slug]").each(function(index, obj) {
+            tableColumnFields[$(obj).data("slug")] = index;
+        });
+
+        opus.prefs.order = [];
+        $.each(data.order_list, function(index, order_entry) {
+            let slug = order_entry.slug;
+            let label = order_entry.label;
+            let isDescending = order_entry.descending;
+            let orderTooltip = (isDescending ? "Change to ascending sort" : "Change to descending sort");
+
+            let removeable = order_entry.removeable;
+            listHtml += "<div class='list-inline-item'>";
+            listHtml += `<span class='badge badge-pill badge-light' data-slug="${slug}" data-descending="${isDescending}">`;
+            if (removeable) {
+                listHtml += "<span class='op-remove-sort' title='Remove metadata field from sort'><i class='fas fa-times-circle'></i></span> ";
+            }
+            listHtml += `<span class='op-flip-sort' title='${orderTooltip}'>`;
+            listHtml += label;
+            listHtml += (isDescending ? `<i class="${pillSortUpArrow} ml-1"></i>` : `<i class="${pillSortDownArrow} ml-1"></i>`);
+            listHtml += "</span></span></div>";
+
+            let fullSlug = slug;
+            if (isDescending) {
+                fullSlug = "-" + slug;
+            }
+            opus.prefs.order.push(fullSlug);
+            if (tableColumnFields[slug] !== undefined) {
+                delete tableColumnFields[slug];
+            }
+        });
+
+        listHtml += `<div class="op-sort-order-add-icon list-inline-item" title="Edit sort by adding metadata fields">`+
+					   `<i class="fas fa-plus"></i>`+
```

Comment:
Bad indent?

Reply:
This is still weirdly indented

Reply:
The indent here is intentional; I am mimicking the html indent in js.